src/components/views/voip/CallView/CallViewHeader.tsx

```diff
@@ -0,0 +1,135 @@
+import WidgetUtils from '../../../../utils/WidgetUtils';
+
+const callTypeTranslationByType: Record<CallType | 'widget', (app?: IApp) => string> = {
+  [CallType.Video]: () => _t("Video Call"),
+  [CallType.Voice]: () => _t("Voice Call"),
+  'widget': (app: IApp) => WidgetUtils.getWidgetName(app),
+};
+
+interface CallViewHeaderProps {
+  pipMode: boolean;
+  type: CallType | 'widget';
+  callRooms?: Room[];
+  app?: IApp;
+  onPipMouseDown: (event: React.MouseEvent<Element, MouseEvent>) => void;
+}
+
+const onRoomAvatarClick = (roomId: string) => {
+    dis.dispatch({
+    action: 'view_room',
+    room_id: roomId,
+    });
+};
+
+const onFullscreenClick = () => {
+    dis.dispatch({
+    action: 'video_fullscreen',
+    fullscreen: true,
+    });
+};
+
+const onExpandClick = (roomId: string) => {
+    dis.dispatch({
+    action: 'view_room',
+    room_id: roomId,
+    });
+};
+
+type CallControlsProps = Pick<CallViewHeaderProps, 'pipMode' | 'type'> & {
+  roomId: string;
+};
+function CallControls({ pipMode = false, type, roomId }: CallControlsProps) {
+    return <div className="mx_CallView_header_controls">
+        { (pipMode && type === CallType.Video) &&
+      <div className="mx_CallView_header_button mx_CallView_header_button_fullscreen"
+          onClick={onFullscreenClick}
+          title={_t("Fill Screen")}
+      /> }
+        { pipMode && <div className="mx_CallView_header_button mx_CallView_header_button_expand"
+            onClick={() => onExpandClick(roomId)}
+            title={_t("Return to call")}
+        /> }
+    </div>;
+}
+function SecondaryCallInfo({ callRoom }: {callRoom: Room}) {
+    return <span className="mx_CallView_header_secondaryCallInfo">
+        <AccessibleButton element='span' onClick={() => onRoomAvatarClick(callRoom.roomId)}>
+            <RoomAvatar room={callRoom} height={16} width={16} />
+            <span className="mx_CallView_secondaryCall_roomName">
+                { _t("%(name)s on hold", { name: callRoom.name }) }
+            </span>
+        </AccessibleButton>
+    </span>;
+}
+
+function getAvatarBasedOnRoomType(roomOrWidget: Room | IApp) {
+    if (roomOrWidget instanceof Room) {
+        return <RoomAvatar room={roomOrWidget} height={32} width={32} />;
+    } else if (!isUndefined(roomOrWidget)) {
+        return <WidgetAvatar app={roomOrWidget} height={32} width={32} />;
+    }
+    return null;
+}
+
+export const CallViewHeader: React.FC<CallViewHeaderProps> = ({
+  type,
+  pipMode = false,
+  callRooms = [],
+  app,
+  onPipMouseDown,
+}) {
+    const [callRoom, onHoldCallRoom] = callRooms;
+    const callTypeText = callTypeTranslationByType[type](app);
+    const avatar = getAvatarBasedOnRoomType(callRoom ?? app);
+    const callRoomName = type === 'widget' ? callTypeText : callRoom.name;
+    const roomId = app ? app.roomId : callRoom.roomId;
+    if (!pipMode) {
+        return <div className="mx_CallView_header">
+            <div className="mx_CallView_header_phoneIcon" />
```

Comment:
This class doesn't exist anymore, we should be using `mx_CallView_header_callTypeIcon` and the related classes.